xskillscore/core/comparative.py

```diff
@@ -0,0 +1,143 @@
+        alpha : float
+            significance level for random walk.
+        categorical : bool, optional
+            If ``True``, the winning forecast is only rewarded a point if it exactly
+            equals the observation. If False, use metric to compare forecast# with
+            observation.
+        metric : callable, optional
+            metric to compare forecast# and observation if metric is not None. If
+            metric is None, assume that forecast# have been compared observation before
+            using ``sign_test``. If categorical is True, metric is ignored. Also allows
+            strings to be convered to ``xskillscore.{metric}``. Defaults to None.
+        orientation : str
+            Which skill values correspond to better skill? Smaller values (negative) or
+            Larger values (positive). Defaults to 'negative'. Ignored if metric is None
+            or categorical.
+
+        Returns
+        -------
+        xarray.DataArray or xarray.Dataset reduced by dim containing the sign test and
+            confidence as new dimension results:
+
+                * ``sign_test``: positive (negative) number shows how many times over
+                    ``dim`` ``forecast1`` is better (worse) than ``forecast2``.
+                * ``confidence``: Positive boundary for the random walk at significance
+                    ``alpha``.
+
+
+        Examples
+        --------
+        >>> f1 = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> f2 = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> o = xr.DataArray(np.random.normal(size=(30)),
+        ...      coords=[('time', np.arange(30))])
+        >>> st = sign_test(f1, f2, o, dim='time')
+        >>> st.sel(results='sign_test').plot()
+        >>> st.sel(results='confidence').plot(c='gray')
+        >>> (-1*st.sel(results='confidence')).plot(c='gray')
+
+        References
+        ----------
+            * DelSole, T., & Tippett, M. K. (2016). Forecast Comparison Based on Random
+              Walks. Monthly Weather Review, 144(2), 615–626. doi: 10/f782pf
+    """
+    # make sure metric is a callable
+    if metric is not None:
+        if isinstance(metric, str):
+            import xskillscore as xs
+
+            if hasattr(xs, metric):
+                metric = getattr(xs, metric)
+            else:
+                raise ValueError(
+                    f'xskillscore.metric could not be derived from {metric}.'
+                )
+        elif not callable(metric):
+            raise ValueError(
+                f'metric needs to be a function/callable or None, found {type(metric)}'
+            )
+
+    if metric:
+        if observation is not None:
+            # Compare the forecasts and observation using metric
+            diff1 = metric(forecast1, observation, dim=[])
```

Comment:
true. now `observation, forecast1, forecast2`

Reply:
Sorry, this is not what I was meaning. 

I think the signature for this function should be as you had it before, because I think we want `observation` to be optional:
```python
def sign_test(forecast1, forecast2, 
              observation=None, dim=None, metric=None, orientation="negative", alpha=0.05):
```
But, the order needs to be reversed on lines 108 and 109

Reply:
now I see what you mean.

I also see a mismatch in the args. either 
1. we make `sign_test` aligned with the `metrics` by having `obs` first. The user could still insert `obs=None`
or
2. we keep `obs` by default `None` but then need to have `obs` third.

I am in favor of 1.

Reply:
although `sign_test(None,fct1,fct2)` reads a bit weird, `sign_test(observation=None,forecast1=fct1,forecast2=fct2)` reads nicer.

Reply:
if you prefer the `sign_test(fct1,fct2,obs)` ordering I will go back to that.

Reply:
I think I prefer option 2 for the reason you give above (ie `sign_test(None, fcst1, fcst2)` is odd).

I personally don't think it's an issue that the ordering has changed for this metric relative to the others in xskillscore because it's for comparing two forecasts.

Reply:
thoughts @bradyrx @raybellwaves

Reply:
I added a thumbs up there.

Reply:
Ok. Agreed.